File: turing-backend/orchestrator/src/tools/policy.ts
import type { ToolPolicy } from "@turing/shared-types";

const POLICIES = new Map<string, ToolPolicy>([
  ["system.health", "safe"],
  ["system.time", "safe"],
  ["system.echo", "safe"],
  ["system.info", "safe"],
  ["files.list", "safe"],
  ["files.search", "safe"],
  ["files.read", "safe"],
  ["files.create", "approval_required"],
  ["files.update", "approval_required"],
  ["files.delete", "disabled"],
  ["files.move", "disabled"]
]);

export function getToolPolicy(toolName: string): ToolPolicy | undefined {
  return POLICIES.get(toolName);
}

File: turing-backend/orchestrator-go/internal/service/tools/policy.go
package tools

type Policy string

const (
	PolicySafe             Policy = "safe"
	PolicyApprovalRequired Policy = "approval_required"
	PolicyDisabled         Policy = "disabled"
)

var policies = map[string]Policy{
	"system.time":   PolicySafe,
	"system.health": PolicySafe,
	"system.echo":   PolicySafe,
	"files.create":  PolicyApprovalRequired,
	"files.update":  PolicyApprovalRequired,
}

func GetPolicy(toolName string) (Policy, bool) {
	policy, ok := policies[toolName]
	return policy, ok
}
